reconcile/openshift_saas_deploy.py

```diff
@@ -100,7 +102,13 @@ def slack_notify(
         message += f" triggered by _{trigger_integration}_"
     if in_progress and skip_successful_notifications:
         message += ". There will not be a notice for success."
-    slack.chat_post_message(message)
+    try:
+        slack.chat_post_message(message)
+    except SlackApiError as e:
+        if e.response["error"] == "channel_not_found":
+            logging.error(f"Slack API says can't find channel {slack.channel}")
+        # raise up to let someone else handle the exception
+        raise e
```

Comment:
this already happens within `chat_post_message`, no?

Reply:
Judging from the output in the Jira card I am not so sure that it is.

Reply:
There are some error handings in https://github.com/app-sre/qontract-reconcile/blob/59c00eb23f40ec94e1f8383b62e9f9211cfa890a/reconcile/utils/slack_api.py#L237, can move this case there too.

Also need to update test case to reflect the correct error type https://github.com/app-sre/qontract-reconcile/blob/59c00eb23f40ec94e1f8383b62e9f9211cfa890a/reconcile/test/utils/test_slack_api.py#L197

Btw to rethrow the exception, should just use `raise` instead of `raise e` to keep the original stack trace.

Reply:
> There are some error handings in
> 
> https://github.com/app-sre/qontract-reconcile/blob/59c00eb23f40ec94e1f8383b62e9f9211cfa890a/reconcile/utils/slack_api.py#L237
> 
> , can move this case there too.

Why would `slack_api.py` care about sending a warning about a channel not existing? It seems like that should be the domain of the user of the API.

> Btw to rethrow the exception, should just use `raise` instead of `raise e` to keep the original stack trace.

I'll make this change.